docker/docker_client.go

```diff
@@ -42,17 +42,17 @@ type dockerClient struct {
 	wwwAuthenticate string // Cache of a value set by ping() if scheme is not empty
 	scheme          string // Cache of a value returned by a successful ping() if not empty
 	client          *http.Client
+	signatureBase   signatureStorageBase
 }
 
 // newDockerClient returns a new dockerClient instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
-func newDockerClient(ctx *types.SystemContext, refHostname string) (*dockerClient, error) {
-	var registry string
-	if refHostname == dockerHostname {
+// forWriting specifies whether the client will be used for "write" access (in particular passed to lookaside.go:toplevelFromSection)
+func newDockerClient(ctx *types.SystemContext, ref dockerReference, forWriting bool) (*dockerClient, error) {
```

Comment:
> can we name this just `write bool` and document that it'll be used _for writing_?

I can’t see that having a short name + “for writing” documentation is any shorter or easier to understand than a self-documenting `forWriting`, but, sure, changed.

> Maybe `mode` or `mask` and have other values to maybe be future proof?

I don’t want to get all Design Patterns on this; the underlying configuration file syntax is _not_ future proof that way, and that is the ABI commitment; even if we turned `mode` into an interface we wouldn’t know what to do with it in `registryNamespace.signatureTopLevel`. The parameter is an internal implementation detail which we can generalize at any time if needed.